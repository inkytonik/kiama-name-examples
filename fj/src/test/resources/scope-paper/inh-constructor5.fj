// ADAPTED FROM inheritance/constructors.spt:call constructor with subtype argument
// FIX: Add fields to C
class A extends Object {
  A() { super(); }
}
class B extends A {
  B() { super(); }
}
class C extends Object {
  A f;
  C(A f){
    super();
    this.f = f;
  }
}
new C(new B())
