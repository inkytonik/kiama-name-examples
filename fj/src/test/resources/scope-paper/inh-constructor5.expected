src/test/resources/scope-paper/inh-constructor5.fj:10:5:info: f def: A
  A f;
    ^
src/test/resources/scope-paper/inh-constructor5.fj:11:7:info: f def: A
  C(A f){
      ^
src/test/resources/scope-paper/inh-constructor5.fj:16:1:info: new C(new  ...: C
new C(new B())
^
src/test/resources/scope-paper/inh-constructor5.fj:16:7:info: new B(): B
new C(new B())
      ^
C
